src/textual/_xterm_parser.py

```diff
@@ -10,6 +10,11 @@
 from ._parser import Awaitable, Parser, TokenCallback
 from ._ansi_sequences import ANSI_SEQUENCES_KEYS
 
+# When trying to determine whether the current sequence is a supported/valid
+# escape sequence, at which length should we give up and consider our search
+# to be unsuccessful?
+_MAX_SEQUENCE_SEARCH_THRESHOLD = 20
```

Comment:
Why was 20 chosen?

Reply:
The largest that I think we could expect to receive (that I'm aware of) is length 14. The existence of the `re_sgr_mouse` regex suggested to me that longer sequences could exist, so I added on some leeway for defensiveness.